src/controllers/stallController.ts

```diff
@@ -462,6 +462,60 @@ async function retrieveFeatureStalls(req: Request, res: Response, next: NextFunc
   }
 }
 
+async function indexCategoryStalls(req: Request, res: Response, next: NextFunction) {
```

Comment:
Is there any reason we need to expose `CategoryStalls` from the API? This seems like an implementation detail. User could send in `categoryIds`, not `categoryStallsIds` and we set the stall's categories based on the given `categoryIds`. A stall's `categoryIds` could be found using the `GET /stalls/:id` route. I think this is also how users are going to get `categoryStallsIds` if the PR is merged in

Reply:
In addition, `sequelize` already provides built-in functions to update `categories` from a `stall`. 

```javascript
// In controllers/stallController.ts
async function updateStallCategories(req: Request, res: Response, next: NextFunction) {
  try {
    const categories: number[] | string[] = req.body.categoryIds;
    const stall = req.stall!;
    await stall.setCategories(categories);
    await stall.reload();
    res.status(200).json(stall);
  } catch (err) {
    next(err);
  }
}

//...

export const updateStallCategories = [retrieveStall, updateStallCategories];

// In routes/stalls.ts
router.put('/:id/setCategories', auth, stallController.updateStallCategories);
```

Reply:
The above function is tested already. You can even send things like `categoryIds: [1, "2", "5"]` and `sequelize` will settle for you.

Reply:
I see. Will implement it, thanks!